default/scripting/ship_parts/ShortRange/SR_TENTACLE.focs.txt

```diff
@@ -3,6 +3,30 @@ Part
     description = "SR_TENTACLE_DESC"
     class = ShortRange
     damage = 5
+    shots = 2
+    combatTargets = And [
+        (CombatBout > 1)
+        [[COMBAT_TARGETS_VISIBLE_ENEMY]]
+        OrderedAlternativesOf [
+           And [
```

Comment:
I am not familiar with scripting language and the combat process.  Could you explain to me a little about `OrderedAlternativesOf`?

According to the description, ships should be attacked first, but when CombatBout is 2 fighters is a priority.

How CombatBout is decremented?  When all ships are destroyed or unconditionally?

Reply:
The OrderedAlternativesOf condition matches all objects that match the first subcondition, unless nothing matches, in which case it moves to the next, and keeps trying subconditions until something is matched. From the code comment:
````
    // OrderedAlternativesOf [ A B C ] matches all candidates which match the topmost condition.
    // If any candidate matches A, then all candidates that match A are matched,
    // or if no candidate matches A but any candidate matches B, then all candidates that match B are matched,
    // or if no candidate matches A or B but any candidate matches C, then all candidates that match C are matched.
    // If no candidate matches A, B, or C, then nothing is matched.
    //
    // Not OrderedAlternativesOf [ A B C ] finds the topmost condition which has matches and then matches its non-matches.
    // If any candidate matches A, then all candidates that do not match A are matched,
    // or if no candidates match A but any candidate matches B, then all candidates that do not match B are matched,
    // or if no candidate matches A or B but any candidate matches C, then all candidates that do not match C are matched.
    // If no candidate matches A, B, or C, then all candidates are matched.
````

Every main game turn, there is a new combat independent of any previous combats in a system, and the CombatBout is reset to 1.

Reply:
> I am not familiar with scripting language and the combat process. Could you explain to me a little about `OrderedAlternativesOf`?

There is also an entry in [FOCS_Scripting_Details](https://www.freeorion.org/index.php/FOCS_Scripting_Details) in the wiki.
From combat bout 3 on ships are primarily targetted . Only if there are no ship enemies, tentacles attack fighters or planets.

> 
> How CombatBout is decremented?

Each battle has (usually) four combat bouts. First bout 1, second bout 2,... You can see this in the combat reports.


> 
> According to the description, ships should be attacked first, but when CombatBout is 2 fighters is a priority.

It also sais: "Tentacles start to destroy fighters when they come in range in [[SHORT_RANGE_COMBAT_ROUND]]."

I introduce here a design/fluff model of "shrinking distances". First bout is "long" distance, second bout is "short" distance, and starting from third bout it is "close". Fighters are supposed to closer to the ships (they shoot from inside the ship shields, fluff said), so in second bout the "close" distance attacking tentacle can not attack ships, but it can reach the fighters which moved already closer. Note this is not done consistently for other weapons (jaws cant attack fighters in bout two), as the tentacles can "stretch". Mostly adding flavor and some utility.

edit1: use the right quote

Reply:
Thank you for the explanation.  I think I got an idea.  

I tried to rephrase the description in the string tables.  I try to make it a story rather than a list of facts.  In the evening my brain not good at interpreting facts.